tests/test_poly.py

```diff
@@ -0,0 +1,287 @@
+import numpy
+import piff
+PolynomialsTypes = piff.POLYNOMIAL_TYPES.keys()
+
+
+
+def test_poly_indexing():
+    #Some indexing tests for a polynomial up to order 3
+    N = 3
+    interp = piff.Polynomial([N])
+
+    #We expect there to be these coefficients:
+    # x^0 y^0   1
+
+    # x^0 y^1   2
+    # x^1 y^0   3
+
+    # x^0 y^2   4
+    # x^1 y^1   5 
+    # x^2 y^0   6
+
+    # x^0 y^3   7
+    # x^1 y^2   8
+    # x^2 y^1   9
+    # x^3 y^0   10 
+
+    #Check that we have the indices we expect
+    assert interp.indices[0] == [
+        (0,0),
+        (0,1), (1,0),
+        (0,2), (1,1), (2,0),
+        (0,3), (1,2), (2,1), (3,0)
+    ]
+    assert interp.nvariables[0]==10
+
+    #check the packing then unpacking a 
+    packed = numpy.random.uniform(size=interp.nvariables[0])
+    unpacked = interp.unpack_coefficients(0,packed)
+    packed_test = interp.pack_coefficients(0,unpacked)
+
+    #Check that the shape is 4*4 in the unpacked (because we
+    #want space for all the terms), and that we can unpack and
+    #repack successfully.
+    assert (packed==packed_test).all()
```

Comment:
numpy.testing has some nicer ways to write tests involving numpy arrays.
```
numpy.testing.assert_array_equal(packed_test, packed)
```
Then for things that don't need to be exact, there are `assert_almost_equal` and similar things that tend to be useful.